File: src/procedures/FillStripInterpolatedProcedure.java
package procedures;

import java.awt.Color;

import interpolation.Interpolation;
import interpolation.InterpolationType;
import led.ProcedureBundleFields;
import led.LEDDataBundle;
import led.LEDStripManager;

public class FillStripInterpolatedProcedure extends Procedure {

	private int mLitLEDs = 0;
	private float mPpercentage = 0.0f;
	
	private Color mFillColor = Color.BLACK;
	InterpolationType mInterpolationType = InterpolationType.EaseOutBounce;

	public FillStripInterpolatedProcedure(LEDDataBundle _bundle) {
		super(_bundle);
		
		mFillColor = _bundle.colorPrimary.toSystemColor();
	}

	@Override
	public void start() {
		
	}
	
	@Override
	void update() {
		mPpercentage = mStep / (float)LEDStripManager.LED_COUNT;
		mStep++;
		
		mLitLEDs = Math.min(300, (int) (Interpolation.getInterpolationValue(mPpercentage, mInterpolationType) * 300.0f));
		
		System.out.println(mLitLEDs);
		
		mStrip.setAllPixels(Color.BLACK);
		mStrip.setArea(0, mLitLEDs, mFillColor);
		
		if (mStep > LEDStripManager.LED_COUNT) {
			mStrip.mProcContainer.removeCurrentProcedure();
			finishProcedure();
		}
	}
}

File: src/main/java/procedures/FillStripInterpolatedProcedure.kt
package procedures

import com.github.iamniklas.interpolation.Interpolation.getInterpolationValue
import com.github.iamniklas.interpolation.InterpolationType
import com.github.iamniklas.liocore.led.LEDDataBundle
import com.github.iamniklas.liocore.led.LEDStripManager
import com.github.iamniklas.liocore.led.colorspace.ColorRGB
import com.github.iamniklas.liocore.procedures.Procedure

class FillStripInterpolatedProcedure(_bundle: LEDDataBundle) : Procedure(_bundle) {
    private var mLitLEDs = 0
    private var mPpercentage = 0.0f
    private var mFillColor = ColorRGB.black.toSystemColor()
    var mInterpolationType = InterpolationType.EaseInOutExpo
    override fun start() {}
    public override fun update() {
        mPpercentage = step / LEDStripManager.LED_COUNT.toFloat()
        step++
        mLitLEDs = Math.min(300, (getInterpolationValue(mPpercentage, mInterpolationType) * 300.0f).toInt())

        //println(mLitLEDs)

        strip!!.setAllPixels(ColorRGB.black.toSystemColor())
        strip!!.setArea(0, mLitLEDs, mFillColor)
        if (step > LEDStripManager.LED_COUNT) {
            strip!!.procContainer.removeCurrentProcedure()
            finishProcedure()
        }
    }

    init {
        mFillColor = _bundle.colorPrimary!!.toSystemColor()
    }
}